### PR TITLE
Small improvements to `debug profile`

### PR DESCRIPTION
**Pull request by kubouch**
# Description

1. With the `-l` flag, `debug profile` now collects files and line numbers of profiled pipeline elements
profiler_lines

2. Error from the profiled closure will be reported instead of silently ignored.
profiler_lines_error

# User-Facing Changes

New `--lines(-l)` flag to `debug profile`. The command will also fail if the profiled closure fails, so technically it is a breaking change.

# Tests + Formatting

# After Submitting

### CODE REVIEW

File: crates/nu-protocol/src/debugger/profiler.rs [hunk 1]
```diff
@@ -262,6 +267,23 @@ fn expr_to_string(engine_state: &EngineState, expr: &Expr) -> String {
     }
 }
 
+// Find a file name and a line number (indexed from 1) of a span
+fn find_file_of_span(engine_state: &EngineState, span: Span) -> Option<(&str, usize)> {
+    for file in engine_state.files() {
+        if file.covered_span.start < span.start && file.covered_span.end > span.start {
```
Comment (IanManske) [inline crates/nu-protocol/src/debugger/profiler.rs:273]:
```suggestion
        if file.covered_span.contains_span(span) {
```

File: crates/nu-protocol/src/debugger/profiler.rs [hunk 2]
```diff
@@ -262,6 +267,23 @@ fn expr_to_string(engine_state: &EngineState, expr: &Expr) -> String {
...
+            // count the number of lines between file start and the searched span start
+            let chunk =
+                engine_state.get_span_contents(Span::new(file.covered_span.start, span.start));
+            let nlines = chunk.lines().count();
+            let line_num = if nlines <= 1 { 1 } else { nlines + 1 };
```
Comment (IanManske) [inline crates/nu-protocol/src/debugger/profiler.rs:278]: Not sure if this is correct but ok.

**Review by IanManske — COMMENTED** [review]
Thanks, nice addition!

File: crates/nu-protocol/src/debugger/profiler.rs (hunk 2 above)
Comment (kubouch) [reply crates/nu-protocol/src/debugger/profiler.rs:278]: I noticed that with only one line, the `nlines + 1` would incorrectly give 2 as the line number, so I special-cased it. This seems to work correctly on regular inputs and files, but I didn't test it thoroughly. There might be some edge cases which we can patch as we discover them.

Comment (IanManske) [reply crates/nu-protocol/src/debugger/profiler.rs:278]: I think this should work? Let me know if anything looks wrong.
```suggestion
            // account for leading part of current line being counted as a separate line
            let line_num = if chunk.last() == Some(&b'\n') {
                nlines + 1
            } else {
                nlines
            };

            // first line has no previous line, clamp up to `1`
            let line_num = usize::max(line_num, 1);
```

File: crates/nu-protocol/src/debugger/profiler.rs [hunk 3]
```diff
@@ -277,6 +299,16 @@ fn collect_data(
         "parent_id" => Value::int(parent_id.0 as i64, profiler_span),
     };
 
+    if profiler.collect_lines {
+        if let Some((fname, line_num)) = find_file_of_span(engine_state, element.element_span) {
+            row.push("file", Value::string(fname, profiler_span));
+            row.push("line_num", Value::int(line_num as i64, profiler_span));
+        } else {
+            row.push("file", Value::nothing(profiler_span));
+            row.push("line_num", Value::nothing(profiler_span));
```
Comment (IanManske) [inline crates/nu-protocol/src/debugger/profiler.rs:308]: One last thing, can we rename this column to just `line`?
```suggestion
            row.push("line", Value::nothing(profiler_span));
```

Comment (IanManske) [reply crates/nu-protocol/src/debugger/profiler.rs:308]: I made the suggestion wrong, there's also another `line_num` above.

**Comment by kubouch** [comment]
Thanks for fixing my sloppy coding 😅

File: crates/nu-protocol/src/debugger/profiler.rs [hunk 4]
```diff
@@ -277,6 +307,16 @@ fn collect_data(
         "parent_id" => Value::int(parent_id.0 as i64, profiler_span),
     };
 
+    if profiler.collect_lines {
+        if let Some((fname, line_num)) = find_file_of_span(engine_state, element.element_span) {
+            row.push("file", Value::string(fname, profiler_span));
+            row.push("line_num", Value::int(line_num as i64, profiler_span));
```
Comment (IanManske) [inline crates/nu-protocol/src/debugger/profiler.rs:313]: Sorry I missed this spot
```suggestion
            row.push("line", Value::int(line_num as i64, profiler_span));
```

**Pull request merged**